tensorflow/core/nccl/nccl_manager.cc

```diff
@@ -16,15 +16,32 @@ limitations under the License.
 
 #include <utility>
 
-#ifdef GOOGLE_CUDA
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 
 #include "tensorflow/core/lib/core/refcount.h"
 #include "tensorflow/core/lib/core/threadpool.h"
-#include "tensorflow/core/platform/cuda.h"
 #include "tensorflow/core/platform/env.h"
+#if GOOGLE_CUDA
+#include "tensorflow/core/platform/cuda.h"
+#elif TENSORFLOW_USE_ROCM
+#include "tensorflow/core/platform/rocm.h"
+#endif
 
 namespace tensorflow {
 
+#if GOOGLE_CUDA
+using se::cuda::ScopedActivateExecutorContext;
```

Comment:
Type alias would be preferred over using declarations here and defines below.

Reply:
How about the following?
```C++
#if GOOGLE_CUDA
using ScopedActivateExecutorContext = se::cuda::ScopedActivateExecutorContext;
#elif TENSORFLOW_USE_ROCM
using ScopedActivateExecutorContext = se::rocm::ScopedActivateExecutorContext;
// Local hipify of cuda symbols
using cudaError_t = hipError_t;
using cudaStream_t = hipStream_t;
const auto cudaGetErrorString = hipGetErrorString;
const auto cudaGetDevice = hipGetDevice;
const auto cudaSetDevice = hipSetDevice;
const auto cudaSuccess = hipSuccess;
#endif
```